sklearn/metrics/_pairwise_distances_reduction/_base.pyx

```diff
@@ -116,11 +116,10 @@ cdef class PairwiseDistancesReduction64:
         by parallelizing computation on the outer loop on chunks of X
         and reduce them.
 
-        This strategy dispatches chunks of Y uniformly on threads.
-        Each thread processes all the chunks of X in turn. This strategy is
-        a sequence of embarrassingly parallel subtasks (the inner loop on Y
-        chunks) with intermediate datastructures synchronisation at each
-        iteration of the sequential outer loop on X chunks.
+        This strategy dispatches chunks of X uniformly on threads.
```

Comment:
Would it be more proper to describe the dispatching of the threads, rather than of the underlying data?
```suggestion
        This strategy dispatches threads on uniform chunks of X.
```

Reply:
That's a good remark. In fact, I think `"static*"` must be prefered over `"uniform*"`.

If one were to describe the tasks using [OpenMP execution model](https://www.openmp.org/spec-html/5.0/openmpse3.html) as a reference, I think one would say that "tasks are statically dispatched on threads", where in this case (i.e. `_parallel_on_X`), "a task consists for a given chunk of X in sequentially computing and reducing the distance matrices between vectors of this chunk and vectors of a chunk of Y, for all chunk of Y".

Hence, what do you think of this rephrasing?
```suggestion
        This strategy dispatches tasks statically on threads.
        Each task processes one and only one chunk of X,
        computing and reducing distances matrices between
        vectors of this chunk and vectors of all chunks of Y,
        one chunk of Y at a time.
```

Reply:
Yes I think that phrasing is much improved! Minor nit would be replacing `one and only one` with `exactly one` but that's just personal preference -- the actual content is great.

Reply:
Good catch. There you go!
```suggestion
        This strategy dispatches tasks statically on threads.
        Each task processes exactly only one chunk of X,
        computing and reducing distances matrices between
        vectors of this chunk and vectors of all chunks of Y,
        one chunk of Y at a time.
```